PsfLauncher/main.cpp

```diff
@@ -550,54 +618,84 @@ ErrorInformation CheckIfPowershellIsInstalled(bool& isPowershellInstalled) noexc
     return {};
 }
 
+ErrorInformation CheckIfShouldRun(bool shouldRunOnce, bool &canScriptRun)
+{
+    canScriptRun = true;
+    if (shouldRunOnce)
+    {
+        std::wstring runOnceSubKey = L"SOFTWARE\\";
+        runOnceSubKey.append(psf::current_package_full_name());
+        runOnceSubKey.append(L"\\runOnce");
+
+        DWORD keyDisposition;
+        wil::unique_hkey registryHandle;
+        LSTATUS createResult = RegCreateKeyExW(HKEY_CURRENT_USER, runOnceSubKey.c_str(), 0, nullptr, REG_OPTION_NON_VOLATILE, KEY_ALL_ACCESS, nullptr, &registryHandle, &keyDisposition);
+
+        if (createResult != ERROR_SUCCESS)
+        {
+            return { L"Error with getting the key to see if the script has already ran. ", (DWORD)createResult };
+        }
+
+        //The user can change the reg key to force the script to run again.
+        //If that is the case make sure hasBeenRan is true.
+        //If false the script can run.
+        if (keyDisposition == REG_OPENED_EXISTING_KEY)

```

Comment:
Seems like this is what you intended with this comment but didn't follow through?

Reply:
I changed the implementation half-way through.  This is no longer allowed.  If they want to re-run the script they need to re-install the app.